### PR TITLE
Fixes silent failure issues

### PR DESCRIPTION
**Pull request by jmuharsky**
In cases where error messages aren't returning a string, Explorer is silently failing, and the widget will continue to appear in a loading state.  This fixes the issue so that "An unexpected error occurred" appears if no error text is provided.

### CODE REVIEW

**Comment by jmuharsky** [comment]
Demo is at https://27-1-dot-perfkit-explorer.googleplex.com

File: client/components/widget/query/query-result-data-service.js [hunk 1]
```diff
@@ -226,26 +228,39 @@ QueryResultDataService.prototype.fetchResults = function(widget) {
...
       } else {
         let rows = response.data.totalRows;
         let size = response.data.totalBytesProcessed;
         let time = response.data.elapsedTime;
-        let job = response.data.jobReference.jobId;
+        let jobReference = response.data.jobReference;
 
-        widget.state().datasource.job_id = job;
+        if (goog.isDefAndNotNull(jobReference)) {
+          widget.state().datasource.job_id = jobReference.jobId;
```
Comment (ivansmf) [inline client/components/widget/query/query-result-data-service.js]: Same comment as the other pull request. You need to know the type of request you are getting here to know whether you can ignore this, right?

File: client/components/widget/query/query-result-data-service.js [hunk 2]
```diff
@@ -44,6 +44,8 @@ const WorkQueueService = explorer.components.util.WorkQueueService;
 const WidgetConfig = explorer.models.WidgetConfig;
 
 
+const ERR_UNEXPECTED = 'An unexpected error occurred.';
```
Comment (ivansmf) [inline client/components/widget/query/query-result-data-service.js]: We can make this more obvious - "An error occurred when fetching data from %s, but the http error offered no details", where %s is the data provider or datasource name. (e.g. BigQuery or CloudSQL now, "my data source #12" in the future".

File: client/components/widget/query/query-result-data-service.js (hunk 1 above)
Comment (jmuharsky) [inline client/components/widget/query/query-result-data-service.js]: Not at present; we just don't build the string if the data isn't there, and we do when it is.  This class is a prime example of still needing proper factoring and encapsulation for extensibility, but for now it suppresses errors.  And even if it's BigQuery, there are conditions where it won't return this string (we've had a few failures that occur before the jobid is returned).

File: client/components/widget/query/query-result-data-service.js (hunk 2 above)
Comment (jmuharsky) [inline client/components/widget/query/query-result-data-service.js]: Added prefix text to all error messages (An error occurred when fetching data from FOO), and rewrote ERR_UNEXPECTED to focus on the HTTP response's lack of detail.

**Comment by jmuharsky** [comment]
Cleaned up comments.  PTAL?

**Comment by jmuharsky** [comment]
To the googlebot: I signed the CLA some time ago, and have been the major contributor to this repository for >1 year.  Please verify the request.

**Pull request merged**